include/revng/Pipes/Isolate.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/EarlyFunctionAnalysis/ABIDetectionPass.h"
+#include "revng/FunctionIsolation/InvokeIsolatedFunctions.h"
+#include "revng/FunctionIsolation/IsolateFunctions.h"
+#include "revng/Pipes/RevngPipes.h"
+#include "revng/Pipes/RootKind.h"
+
+namespace pipeline {
+
+struct ABIDetectPipe {
+  static constexpr auto Name = "ABIDetect";
+
+  std::vector<ContractGroup> getContract() const { return {}; }
+
+  void registerPasses(llvm::legacy::PassManager &Manager) {
+    Manager.add(new EarlyFunctionAnalysis::ABIDetectionPass());
+  }
+};
+
+struct IsolatePipe {
+  static constexpr auto Name = "Isolate";
+
+  std::vector<ContractGroup> getContract() const {
+    return { ContractGroup(Root,
+                           Exactness::Exact,
+                           0,
+                           Isolated,
+                           0,
+                           InputPreservation::Preserve) };
```

Comment:
This is not very nice, the indices (`0` and `0`) refer to nothing immediately visible. This is very obscure.
Can we make a `LLVMContractGroup` that has a constructor with those values hardcoded so we can hide this from the user?